frontend/src/features/StartLesson/lessonAPI.ts

```diff
@@ -0,0 +1,23 @@
+const axios = require("axios").default;
+
+export const fetchQuizDetail = async (quizId: number) => {
+  try {
+    const response = await axios.get(
+      `http://127.0.0.1:8000/api/quizzes/${quizId}`
+    );
+    return response.data;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const fetchQuestions = async (quizId: number) => {
+  try {
+    const response = await axios.get(
+      `http://127.0.0.1:8000/api/quizzes/${quizId}/questions`
+    );
+    return response.data;
+  } catch (error) {
+    console.error(error);
+  }
+};
```

Comment:
For axios no need to wrap around try catch blocks since axios has built in functions/callbacks for it using `then()` and `catch()` methods where will be called when its a success `then()` or error `catch()` response. So in this case you can simply just return the axios like so.

Also axios methods like get, post, delete, request, etc. are already synchronous functions so by returning this function itself not need to add async await

```suggestion
export const fetchQuizDetail = (quizId: number) => {
  return axios.get(`http://127.0.0.1:8000/api/quizzes/${quizId}`);
};

export const fetchQuestions = (quizId: number) => {
  return axios.get(`http://127.0.0.1:8000/api/quizzes/${quizId}`);
};
```

more details on their features here:
https://www.npmjs.com/package/axios

Reply:
Just additional information:
so in some cases you just need to call the api and dont need redux's dispatch for it you may call it in the component like so.
```js
fetchQuestions(quizId)
.then(res => {
   // Logic to handle for success here..
})
.catch(err => {
  // Logic to handle for error here..
});